packages/editor/src/operations/branch/createComponentBranchOperation.ts

```diff
@@ -17,10 +17,10 @@ export type CreateComponentBranchOperationContext = {
 };
 
 export class CreateComponentBranchOperation extends BaseBranchOperation<CreateComponentBranchOperationContext> {
-  do(prev: ComponentSchema[]): ComponentSchema[] {
+  do(prev: AppModel): AppModel {
     // gen component id
     if (!this.context.componentId) {
-      this.context.componentId = genId(this.context.componentType, prev);
+      this.context.componentId = prev.genId(this.context.componentType as ComponentType);
```

Comment:
should we narrow the type of context.componentType to ComponentType from string?